erpnext/accounts/report/general_ledger/general_ledger.py

```diff
@@ -373,16 +370,21 @@ def get_data_with_opening_closing(filters, account_details, accounting_dimension
 			if acc_dict.entries:
 				# opening
 				data.append({"debit_in_transaction_currency": None, "credit_in_transaction_currency": None})
-				if filters.get("group_by") != "Group by Voucher":
+				if (not filters.group_by and not filters.voucher_no) or (
+					filters.group_by and filters.group_by != "Group by Voucher"
+				):
 					data.append(acc_dict.totals.opening)
 
 				data += acc_dict.entries
 
 				# totals
-				data.append(acc_dict.totals.total)
+				if not (not filters.get("group_by") and filters.voucher_no):
```

Comment:
Can this be simplified as `if filters.get("group_by") or not filters.voucher_no:`